niriPipe/niriReduce.py

```diff
@@ -1,14 +1,89 @@
 import argparse
 import niriPipe.inttests
+import niriPipe.utils.state
+import logging
+import os
+import pkg_resources
+import json
+import glob
 
 
-def run_main(*args, **kwargs):
-    print("Recieved the following args:")
-    for arg in args:
-        print(arg)
-    print("Recieved the following kwargs:")
-    for arg in kwargs:
-        print("{}: {}".format(arg, kwargs[arg]))
+def _create_logger():
+    root_logger = logging.getLogger(__name__.split('.')[0])
+    root_logger.propagate = False
+    root_logger.setLevel(logging.INFO)
+
+    ch = logging.StreamHandler()
+    formatter = logging.Formatter(
+        '%(asctime)s %(name)s %(levelname)s %(message)s')
+    ch.setFormatter(formatter)
+    root_logger.addHandler(ch)
+
+    module_logger = logging.getLogger(__name__)
+
+    return module_logger
+
+
+module_logger = _create_logger()
+
+
+def run_main(args):
+    """
+    Run a NIRI pipeline.
+    """
+
+    if hasattr(args, 'verbose'):
+        # Set niriPipe root logging level to DEBUG
+        logging.getLogger(__name__.split('.')[0]).setLevel(logging.DEBUG)
+
+    module_logger.info("Starting NIRI pipeline.")
+
+    # Get initial state
+    if hasattr(args, 'config') and args.config:
+        configfile = args.config[0]
+    else:
+        configfile = pkg_resources.resource_filename(
+            'niriPipe',
+            os.path.join('cfg', 'default_config.cfg'))
+    state = niriPipe.utils.state.get_initial_state(
+        obs_name=args.obsID,
+        configfile=configfile
+    )
+    module_logger.debug("Initial state:")
+    module_logger.debug(json.dumps(state, sort_keys=True, indent=4))
+
+    # Create and run finder
+    module_logger.info(
+        "Starting data finder for observation {}".format(
+            state['current_stack']['obs_name']))
+    try:
+        finder = niriPipe.utils.finder.Finder(state)
+        data_table = finder.run()
+    except Exception as e:
+        logging.critical("Datafinder failed!")
+        raise e
+    module_logger.info(
+        "Finder succeeded; found {} files.".format(len(data_table)))
+
+    # Run downloader on found files
+    module_logger.info("Starting downloader.")
+    try:
+        downloader = \
+            niriPipe.utils.downloader.Downloader(state=state, table=data_table)
+        downloader.download_query_cadc()
+    except Exception as e:
+        logging.critical("Downloader failed!")
+        raise e
+    module_logger.info("Downloader succeeded.")
+
+    # Check to see if a "stack" was created.
+    if glob.glob("*_stack.fits"):
+        module_logger.info(
+            "Output stack found: {}".format(glob.glob("*_stack.fits")[0]))
+    else:
+        module_logger.warning("Output stack not found.")
+
```

Comment:
Just a style comment: you could have a `step` variable (`Datafinder`, `Donwloader` etc) and put everything into the same `try/except` and log the failing `step` in `except`.